happytransformer/happy_generation.py

```
@@ -0,0 +1,169 @@
+
+    def __check_default_text_is_val(self, text):
+
+        if not isinstance(text, str):
+            self.logger.error("Please enter a int for the max_length parameter")
+            return False
+        elif not text:
+            self.logger.error("The text input must have at least one character")
+            return False
+        return True
+
+
+    def generate_text(self, text, settings=default_greedy_settings,
+                      min_length=20, max_length=60) -> GenerationResult:
+        """
+        :param text: starting text that the model uses to generate text with.
+        :param method: either one of 1/5 preconfigured methods, or "custom" to indicate custom settings
+        :param custom_settings: if method == "custom", then custom settings may be provided in the form of
+              a dictionary. Refer to the README to see potential parameters to add.
+              Parameters that are not added to the dictionary will keep their default value.
+        :return: Text that the model generates.
+        """
+
+        is_valid = self.__check_default_text_is_val(text)
+
+        if is_valid:
+            settings = self.get_settings(settings)
+            input_ids = self.tokenizer.encode(text, return_tensors="pt")
+            adjusted_min_length = min_length + len(input_ids[0])
+            adjusted_max_length = max_length + len(input_ids[0])
+            output = self.model.generate(input_ids,
+                                         min_length=adjusted_min_length,
+                                         max_length=adjusted_max_length,
+                                         **settings
+                                         )
+            result = self.tokenizer.decode(output[0], skip_special_tokens=True)
+            final_result = self.__gt_post_processing(result, text)
+
+            return GenerationResult(text=final_result)
+
+        else:
+            return GenerationResult(text="")
```

This should really be raising an exception instead of silently failing

---

Good catch. 

I added exceptions and simplified the code